### PR TITLE
fix .writeStylesToDisk()

### PR DESCRIPTION
**Pull request by dutchenkoOleg**
If set `styles` option value with path to the folder that is not exist - getting FS error!

```js
// webpack.config.js
// ...
plugins: [
    new SVGSpritemapPlugin({
        src: './resources/assets/svg/*.svg',
        styles: './path-to/my-folder-that-is-not-exist/spritemap.scss'
    })
],
// ...
```

```bash
Error: ENOENT: no such file or directory,
open 'C:\OSPanel\domains\uzv.loc\path-to\my-folder-that-is-not-exist\spritemap.scss'
```

The problem is here [index.js#L335](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/blob/master/lib/index.js#L335)

Because method `fs.writeFileSync(file, data, enc)`, by specs, expect that you will write _only file_,
it does not create folders in path
https://nodejs.org/docs/latest/api/fs.html#fs_fs_writefilesync_file_data_options

----

I added checking: if directories path to location is not exist, then create it

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,13 @@
...
+
+# IDE
+.idea/
+
+# fs
+.DS_Store
+thumbs.db
+
+# logs
+*.log*
```
Comment (cascornelissen) [inline .gitignore]: These should go in your global gitignore

File: lib/index.js [hunk 2]
```diff
@@ -326,12 +327,25 @@ module.exports = class SVGSpritemapPlugin {
             return;
         }
 
+        // check is location exist
+        const locationIsExist = fs.existsSync(location);
```
Comment (cascornelissen) [inline lib/index.js]: Could you rename this to `locationExists`?

File: lib/index.js [hunk 3]
```diff
@@ -326,12 +327,25 @@ module.exports = class SVGSpritemapPlugin {
...
+        const contents = locationIsExist && fs.readFileSync(location, 'utf8');
         if ( contents === styles ) {
             return;
         }
 
+        // if location type is 'dir'
+        // check if directories path to location is exist
+        // and create its if not
+        if (type === 'dir' && !locationIsExist) {
+            let locationDirname = path.dirname(location);
```
Comment (cascornelissen) [inline lib/index.js]: `const dirname`

**Comment by dutchenkoOleg** [comment]
@cascornelissen thx for quick answer. please review changes

File: .gitignore [hunk 4]
```diff
@@ -7,3 +7,6 @@
 /*.scss
 /*.sass
 /*.less
+
+# IDE
+.idea/
```
Comment (cascornelissen) [inline .gitignore]: These should also go in your global gitignore

Comment (dutchenkoOleg) [reply .gitignore]: mission complete))

**Pull request merged**